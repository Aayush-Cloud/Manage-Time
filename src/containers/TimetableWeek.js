// @flow
import React from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';
import SplashScreen from 'react-native-splash-screen';
import { makeGetSortedAppointmentsPerDay } from '../reducers/appointments';
import TimetableDay from './TimetableDay';
import type { AppointmentsPerDay } from '../reducers/appointments';

type OwnProps = {
  /**
   * If shouldCloseLaunchScreen is true, the launch screen will be closed
   * when this component did mount. This is used to only close the launch
   * screen when the initial screen finished rendering.
   */
  shouldCloseLaunchScreen: boolean,
  week: moment$Moment,
};

type Props = OwnProps & {
  appointmentsPerDay: AppointmentsPerDay,
};

// exported for tests
export class TimetableWeek extends React.Component<Props, {}> {
  componentDidMount() {
    if (this.props.shouldCloseLaunchScreen) {
      SplashScreen.hide();
    }
  }

  render() {
    const { appointmentsPerDay, week } = this.props;
    return (
      <View>
        {appointmentsPerDay.map((appointmentsOfDay, index) => {
          const dateString = week
            .clone()
            .weekday(index)
            .format('dddd, LL');
          return (
            <TimetableDay
              day={dateString}
              appointmentsOfDay={appointmentsOfDay}
              key={dateString}
            />
          );
        })}
      </View>
    );
  }
}

function makeMapStateToProps() {
  const getSortedAppointmentsPerDay = makeGetSortedAppointmentsPerDay();
  const mapStateToProps = (state: ApplicationState, ownProps: OwnProps) => ({
    appointmentsPerDay: getSortedAppointmentsPerDay(state, ownProps.week),
  });
  return mapStateToProps;
}

const ConnectedTimetableWeek: (props: OwnProps) => React$Element<any> = connect(
  makeMapStateToProps,
)(TimetableWeek);

export default ConnectedTimetableWeek;
