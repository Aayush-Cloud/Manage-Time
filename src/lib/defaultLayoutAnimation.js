// @flow
import { LayoutAnimation } from 'react-native';

const animation = LayoutAnimation.create(80, 'linear', 'opacity');

export default animation;
