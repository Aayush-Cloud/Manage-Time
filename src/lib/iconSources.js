// @flow
import Icon from 'react-native-vector-icons/MaterialIcons';
import { toPairs } from 'ramda';
import { metrics } from '../themes';

const icons = {
  close: {
    size: metrics.icons.small,
  },
};

const iconSources = {
  close: undefined,
};

// $FlowFixMe
const iconsLoaded = new Promise(resolve => {
  const iconNamePropsPairs = toPairs(icons);
  Promise.all(
    iconNamePropsPairs.map(([iconName, iconProps]) =>
      Icon.getImageSource(iconName, iconProps.size),
    ),
  ).then(sources => {
    iconNamePropsPairs.forEach(([iconName], index) => {
      iconSources[iconName] = sources[index];
    });
    resolve(true);
  });
});

export { iconSources, iconsLoaded };
