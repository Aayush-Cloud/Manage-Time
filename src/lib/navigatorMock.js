// @flow
export default {
  push: jest.fn(),
  pop: jest.fn(),
  popToRoot: jest.fn(),
  resetTo: jest.fn(),
  showModal: jest.fn(),
  dismissModal: jest.fn(),
  dismissAllModals: jest.fn(),
  showLightBox: jest.fn(),
  dismissLightBox: jest.fn(),
  handleDeepLink: jest.fn(),
  setOnNavigatorEvent: jest.fn(),
  setButtons: jest.fn(),
  setTitle: jest.fn(),
  toggleNavBar: jest.fn(),
};
