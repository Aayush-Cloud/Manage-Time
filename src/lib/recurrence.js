// @flow
export const WEEKLY = 'WEEKLY';
export const BIWEEKLY = 'BIWEEKLY';
