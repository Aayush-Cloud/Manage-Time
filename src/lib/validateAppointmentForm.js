// @flow
import i18n from 'react-native-i18n';

export default function validate(values: any) {
  const errors = {};
  // starttime and endtime errors are displayed in same <Text />
  // so both errors are just put into errors.endtime
  if (values.starttime.isAfter(values.endtime, 'minutes')) {
    errors.endtime = i18n.t('endtime-greater-than-starttime');
  }
  if (values.startdate.isAfter(values.enddate, 'minutes')) {
    errors.enddate = i18n.t('enddate-greater-than-startdate');
  }
  return errors;
}
