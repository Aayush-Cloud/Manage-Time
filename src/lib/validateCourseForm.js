// @flow
import i18n from 'react-native-i18n';

export default function validate(values: any) {
  const errors = {};
  if (!values.name) {
    errors.name = i18n.t('title-is-required');
  }
  if (!values.appointments || !values.appointments.length) {
    errors.appointments = { _error: i18n.t('one-appointment-required') };
  }
  return errors;
}
