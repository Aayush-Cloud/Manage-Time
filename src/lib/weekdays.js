// @flow

export const MON = 'mon';
export const TUE = 'tue';
export const WED = 'wed';
export const THU = 'thu';
export const FRI = 'fri';
export const SAT = 'sat';
export const SUN = 'sun';

const weekdays = [SUN, MON, TUE, WED, THU, FRI, SAT];
export default weekdays;
