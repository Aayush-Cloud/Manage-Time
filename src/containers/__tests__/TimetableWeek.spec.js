// @flow
import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import { WEEKLY } from '../../lib/recurrence';
import { TimetableWeek } from '../TimetableWeek';

const fixedMoment = moment('01-01-2017', 'MM-DD-YYYY').startOf('day');
const appointment1: Appointment = {
  id: 'app1',
  startdate: fixedMoment,
  enddate: fixedMoment.clone().add(1, 'month'),
  starttime: fixedMoment,
  endtime: fixedMoment.clone().add(2, 'hours'),
  course: 'course1',
  recurrence: WEEKLY,
};
const appointment2: Appointment = {
  ...appointment1,
  id: 'app2',
  startdate: fixedMoment.clone().add(1, 'day'),
};
const appointment3: Appointment = {
  ...appointment1,
  id: 'app3',
  starttime: fixedMoment.clone().add(1, 'hours'),
};
const appointmentsPerDay = [
  [appointment1, appointment3],
  [appointment2],
];

const props = {
  shouldCloseLaunchScreen: false,
  week: fixedMoment.startOf('week'),
  appointmentsPerDay,
};

describe('TimetableScreen', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <TimetableWeek {...props} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
