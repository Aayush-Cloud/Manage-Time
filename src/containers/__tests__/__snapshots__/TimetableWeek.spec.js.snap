// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimetableScreen renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View>
    <Connect(TimetableDay)
        appointmentsOfDay={
            Array [
                Object {
                  "course": "course1",
                  "enddate": "2017-01-31T23:00:00.000Z",
                  "endtime": "2017-01-01T01:00:00.000Z",
                  "id": "app1",
                  "recurrence": "WEEKLY",
                  "startdate": "2016-12-31T23:00:00.000Z",
                  "starttime": "2016-12-31T23:00:00.000Z",
                },
                Object {
                  "course": "course1",
                  "enddate": "2017-01-31T23:00:00.000Z",
                  "endtime": "2017-01-01T01:00:00.000Z",
                  "id": "app3",
                  "recurrence": "WEEKLY",
                  "startdate": "2016-12-31T23:00:00.000Z",
                  "starttime": "2017-01-01T00:00:00.000Z",
                },
              ]
        }
        day="Sunday, January 1, 2017"
    />
    <Connect(TimetableDay)
        appointmentsOfDay={
            Array [
                Object {
                  "course": "course1",
                  "enddate": "2017-01-31T23:00:00.000Z",
                  "endtime": "2017-01-01T01:00:00.000Z",
                  "id": "app2",
                  "recurrence": "WEEKLY",
                  "startdate": "2017-01-01T23:00:00.000Z",
                  "starttime": "2016-12-31T23:00:00.000Z",
                },
              ]
        }
        day="Monday, January 2, 2017"
    />
</View>,
  "nodes": Array [
    <View>
      <Connect(TimetableDay)
            appointmentsOfDay={
                  Array [
                        Object {
                          "course": "course1",
                          "enddate": "2017-01-31T23:00:00.000Z",
                          "endtime": "2017-01-01T01:00:00.000Z",
                          "id": "app1",
                          "recurrence": "WEEKLY",
                          "startdate": "2016-12-31T23:00:00.000Z",
                          "starttime": "2016-12-31T23:00:00.000Z",
                        },
                        Object {
                          "course": "course1",
                          "enddate": "2017-01-31T23:00:00.000Z",
                          "endtime": "2017-01-01T01:00:00.000Z",
                          "id": "app3",
                          "recurrence": "WEEKLY",
                          "startdate": "2016-12-31T23:00:00.000Z",
                          "starttime": "2017-01-01T00:00:00.000Z",
                        },
                      ]
            }
            day="Sunday, January 1, 2017"
      />
      <Connect(TimetableDay)
            appointmentsOfDay={
                  Array [
                        Object {
                          "course": "course1",
                          "enddate": "2017-01-31T23:00:00.000Z",
                          "endtime": "2017-01-01T01:00:00.000Z",
                          "id": "app2",
                          "recurrence": "WEEKLY",
                          "startdate": "2017-01-01T23:00:00.000Z",
                          "starttime": "2016-12-31T23:00:00.000Z",
                        },
                      ]
            }
            day="Monday, January 2, 2017"
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <TimetableWeek
        appointmentsPerDay={
                Array [
                        Array [
                          Object {
                            "course": "course1",
                            "enddate": "2017-01-31T23:00:00.000Z",
                            "endtime": "2017-01-01T01:00:00.000Z",
                            "id": "app1",
                            "recurrence": "WEEKLY",
                            "startdate": "2016-12-31T23:00:00.000Z",
                            "starttime": "2016-12-31T23:00:00.000Z",
                          },
                          Object {
                            "course": "course1",
                            "enddate": "2017-01-31T23:00:00.000Z",
                            "endtime": "2017-01-01T01:00:00.000Z",
                            "id": "app3",
                            "recurrence": "WEEKLY",
                            "startdate": "2016-12-31T23:00:00.000Z",
                            "starttime": "2017-01-01T00:00:00.000Z",
                          },
                        ],
                        Array [
                          Object {
                            "course": "course1",
                            "enddate": "2017-01-31T23:00:00.000Z",
                            "endtime": "2017-01-01T01:00:00.000Z",
                            "id": "app2",
                            "recurrence": "WEEKLY",
                            "startdate": "2017-01-01T23:00:00.000Z",
                            "starttime": "2016-12-31T23:00:00.000Z",
                          },
                        ],
                      ]
        }
        week={"2016-12-31T23:00:00.000Z"}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "appointmentsPerDay": Array [
            Array [
              Object {
                "course": "course1",
                "enddate": "2017-01-31T23:00:00.000Z",
                "endtime": "2017-01-01T01:00:00.000Z",
                "id": "app1",
                "recurrence": "WEEKLY",
                "startdate": "2016-12-31T23:00:00.000Z",
                "starttime": "2016-12-31T23:00:00.000Z",
              },
              Object {
                "course": "course1",
                "enddate": "2017-01-31T23:00:00.000Z",
                "endtime": "2017-01-01T01:00:00.000Z",
                "id": "app3",
                "recurrence": "WEEKLY",
                "startdate": "2016-12-31T23:00:00.000Z",
                "starttime": "2017-01-01T00:00:00.000Z",
              },
            ],
            Array [
              Object {
                "course": "course1",
                "enddate": "2017-01-31T23:00:00.000Z",
                "endtime": "2017-01-01T01:00:00.000Z",
                "id": "app2",
                "recurrence": "WEEKLY",
                "startdate": "2017-01-01T23:00:00.000Z",
                "starttime": "2016-12-31T23:00:00.000Z",
              },
            ],
          ],
          "week": "2016-12-31T23:00:00.000Z",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View>
          <Connect(TimetableDay)
                    appointmentsOfDay={
                              Array [
                                        Object {
                                          "course": "course1",
                                          "enddate": "2017-01-31T23:00:00.000Z",
                                          "endtime": "2017-01-01T01:00:00.000Z",
                                          "id": "app1",
                                          "recurrence": "WEEKLY",
                                          "startdate": "2016-12-31T23:00:00.000Z",
                                          "starttime": "2016-12-31T23:00:00.000Z",
                                        },
                                        Object {
                                          "course": "course1",
                                          "enddate": "2017-01-31T23:00:00.000Z",
                                          "endtime": "2017-01-01T01:00:00.000Z",
                                          "id": "app3",
                                          "recurrence": "WEEKLY",
                                          "startdate": "2016-12-31T23:00:00.000Z",
                                          "starttime": "2017-01-01T00:00:00.000Z",
                                        },
                                      ]
                    }
                    day="Sunday, January 1, 2017"
          />
          <Connect(TimetableDay)
                    appointmentsOfDay={
                              Array [
                                        Object {
                                          "course": "course1",
                                          "enddate": "2017-01-31T23:00:00.000Z",
                                          "endtime": "2017-01-01T01:00:00.000Z",
                                          "id": "app2",
                                          "recurrence": "WEEKLY",
                                          "startdate": "2017-01-01T23:00:00.000Z",
                                          "starttime": "2016-12-31T23:00:00.000Z",
                                        },
                                      ]
                    }
                    day="Monday, January 2, 2017"
          />
</View>,
        "_debugID": 2,
        "_renderedOutput": <View>
          <Connect(TimetableDay)
                    appointmentsOfDay={
                              Array [
                                        Object {
                                          "course": "course1",
                                          "enddate": "2017-01-31T23:00:00.000Z",
                                          "endtime": "2017-01-01T01:00:00.000Z",
                                          "id": "app1",
                                          "recurrence": "WEEKLY",
                                          "startdate": "2016-12-31T23:00:00.000Z",
                                          "starttime": "2016-12-31T23:00:00.000Z",
                                        },
                                        Object {
                                          "course": "course1",
                                          "enddate": "2017-01-31T23:00:00.000Z",
                                          "endtime": "2017-01-01T01:00:00.000Z",
                                          "id": "app3",
                                          "recurrence": "WEEKLY",
                                          "startdate": "2016-12-31T23:00:00.000Z",
                                          "starttime": "2017-01-01T00:00:00.000Z",
                                        },
                                      ]
                    }
                    day="Sunday, January 1, 2017"
          />
          <Connect(TimetableDay)
                    appointmentsOfDay={
                              Array [
                                        Object {
                                          "course": "course1",
                                          "enddate": "2017-01-31T23:00:00.000Z",
                                          "endtime": "2017-01-01T01:00:00.000Z",
                                          "id": "app2",
                                          "recurrence": "WEEKLY",
                                          "startdate": "2017-01-01T23:00:00.000Z",
                                          "starttime": "2016-12-31T23:00:00.000Z",
                                        },
                                      ]
                    }
                    day="Monday, January 2, 2017"
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TimetableWeek
    appointmentsPerDay={
        Array [
            Array [
              Object {
                "course": "course1",
                "enddate": "2017-01-31T23:00:00.000Z",
                "endtime": "2017-01-01T01:00:00.000Z",
                "id": "app1",
                "recurrence": "WEEKLY",
                "startdate": "2016-12-31T23:00:00.000Z",
                "starttime": "2016-12-31T23:00:00.000Z",
              },
              Object {
                "course": "course1",
                "enddate": "2017-01-31T23:00:00.000Z",
                "endtime": "2017-01-01T01:00:00.000Z",
                "id": "app3",
                "recurrence": "WEEKLY",
                "startdate": "2016-12-31T23:00:00.000Z",
                "starttime": "2017-01-01T00:00:00.000Z",
              },
            ],
            Array [
              Object {
                "course": "course1",
                "enddate": "2017-01-31T23:00:00.000Z",
                "endtime": "2017-01-01T01:00:00.000Z",
                "id": "app2",
                "recurrence": "WEEKLY",
                "startdate": "2017-01-01T23:00:00.000Z",
                "starttime": "2016-12-31T23:00:00.000Z",
              },
            ],
          ]
    }
    week={"2016-12-31T23:00:00.000Z"}
/>,
}
`;
